### ISSUE TITLE
UTF-8 confuses medea about value length

### ISSUE THREAD

**Issue by Ivshti** [medea/medea#48]
Hello, 

It seems that Medea prepends the length of every value in the data file.
However, if the value contains UTF-8 characters, the string length (by bytes) when reading/seeking files varies from what Medea perceives as length which is the String.length property.

As a result, every time values with UTF-8 characters in them are .put in a Medea store, they aren't retrieved back in their entirety when doing a .get.

This is quite a critical issue.
I'll submit a PR if I find it and fix it.

Thanks,
Ivo

**Comment by Ivshti** [comment]
@kevinswiber could this have re-surfaced in medea 1.x.x? I encountered issues with utf-8 encoding, not sure if it's the same issue or not.

```
�-тест (2002).mp4{"uninteresting":true}
^
SyntaxError: Unexpected token �
    at Object.parse [as decode] (native)
    at Object.decodeValue (/Users/ivogeorgiev/stremio-local-files/node_modules/level-sublevel/node_modules/levelup/lib/codec.js:53:35)
    at /Users/ivogeorgiev/stremio-local-files/node_modules/level-sublevel/nut.js:121:34
    at /Users/ivogeorgiev/stremio-local-files/node_modules/medeadown/medeadown.js:127:5
    at /Users/ivogeorgiev/stremio-local-files/node_modules/medeadown/node_modules/medea/medea.js:544:7
    at FSReqWrap.wrapper [as oncomplete] (fs.js:527:17)
```

Obviously medea returns corrupted results in this case

**Comment by kevinswiber** [comment]
@Ivshti Thanks for reporting.  I'm having trouble reproducing this.  Can you provide a test case?

**Comment by Ivshti** [comment]
@kevinswiber unfortunately I lost my test case, but I do know that it doesn't happen on 0.x.x releases

Is there something we can do to deduce where that corrupting case comes from?

I also think that last I tried it happened w/o utf8 symbols, then I rolled back to 0.x.x and the problem went away.

Is it possible it's fixed by one of the latest commits? If I manage to set-up my test case again I'll help
